src/cache/cache.rs

```diff
@@ -125,7 +125,19 @@ impl CacheWrite {
     pub fn put_object<T>(&mut self, name: &str, from: &mut T, mode: Option<u32>) -> Result<()>
         where T: Read,
     {
-        let opts = FileOptions::default().compression_method(CompressionMethod::Deflated);
+        let method = if let Ok(ref method) = env::var("SCCACHE_COMPRESSION_METHOD") {
+            match method.to_lowercase().trim() {
+                "deflate" => CompressionMethod::Deflated,
+                "none" => CompressionMethod::Stored,
+                _ => {
+                    warn!("Unknown compression method {} using 'deflate'.", method);
```

Comment:
I am not entirely sure what you mean by silent failure, could you clarify? My thinking is that it is logging a warning and falling back to deflate in the case of typos. If it is to be a hard error (prevents caching), I would think the proper place to do this check would be startup. It doesn't seem that sccache makes those kinds of configuration checks yet.